libdevice/include/sanitizer_utils.hpp

```diff
@@ -28,7 +28,11 @@ static inline size_t LocalLinearId() {
 
 // For GPU device, each sub group is a hardware thread
 inline size_t SubGroupLinearId() {
-  return __spirv_BuiltInGlobalLinearId / __spirv_BuiltInSubgroupSize;
+  const size_t NumSGPerWG =
+      (__spirv_BuiltInWorkgroupSize.x * __spirv_BuiltInWorkgroupSize.y *
+       __spirv_BuiltInWorkgroupSize.z) /
+      __spirv_BuiltInSubgroupSize;
```

Comment:
Will this go wrong if wg_size is not divisible by sg_size? Maybe we need to take `ceil()` here?